pkg/controller/bucket/bucket_controller.go

```diff
@@ -99,18 +100,32 @@ func (bl *bucketListener) Add(ctx context.Context, obj *v1alpha1.Bucket) error {
 	}
 
 	req := osspec.ProvisionerCreateBucketRequest{
-		BucketName: obj.Spec.BucketRequest.Name,
+		BucketContext: map[string]string{},
+	}
+	if len(obj.Spec.Parameters["BucketInstanceName"]) > 0 {
+		req.BucketName = obj.Spec.Parameters["BucketInstanceName"]
+	} else if len(obj.Spec.Parameters["BucketPrefix"]) > 0 {
+		req.BucketContext["BucketPrefix"] = obj.Spec.Parameters["BucketPrefix"]
+	}
+
+	anonAccessMode := obj.Spec.Parameters["AnonymousAccessMode"]
+	if len(anonAccessMode) > 0 {
+		req.AnonymousBucketAccessMode = osspec.ProvisionerCreateBucketRequest_AnonymousBucketAccessMode(osspec.ProvisionerCreateBucketRequest_AnonymousBucketAccessMode_value[anonAccessMode])
 	}
 
 	switch obj.Spec.Protocol.Name {
 	case v1alpha1.ProtocolNameS3:
 		req.Region = obj.Spec.Protocol.S3.Region
+		req.BucketContext["Version"] = obj.Spec.Protocol.S3.Version
+		req.BucketContext["SignatureVersion"] = string(obj.Spec.Protocol.S3.SignatureVersion)
 	case v1alpha1.ProtocolNameAzure:
+		req.BucketContext["StorageAccount"] = obj.Spec.Protocol.AzureBlob.StorageAccount
 	case v1alpha1.ProtocolNameGCS:
+		req.BucketContext["ServiceAccount"] = obj.Spec.Protocol.GCS.ServiceAccount
+		req.BucketContext["PrivateKeyName"] = obj.Spec.Protocol.GCS.PrivateKeyName
+		req.BucketContext["ProjectID"] = obj.Spec.Protocol.GCS.ProjectID
```

Comment:
There is a contract we are building here. This is a passthrough from the BR->B-> here. Currently that means BR has RequestedProtocol. The other values expected may come from Parameters from BC. Not sure how GCS.ServiceAccount, GCS.PrivateKeyName and GCS.ProjectID get populated. Same for other protocols.

Reply:
This will have to be copied in the central controller from the BucketClass I expect.  The B needs to have everything required for the sidecar to function.